snapcraft/plugins/kernel.py

```diff
@@ -420,7 +420,15 @@ def _do_check_config(self, builtin, modules):
         if missing:
             warn = '\n{}\n'.format(msg)
             for opt in missing:
-                warn += '{}\n'.format(opt)
+                ver = ''
+                if opt == 'CONFIG_CC_STACKPROTECTOR_STRONG':
+                    ver = '(4.1.x and later versions only)'
+                elif opt == 'CONFIG_DEVPTS_MULTIPLE_INSTANCES':
+                    ver = '(4.8.x and earlier versions only)'
+                warn += '{}'.format(opt)
+                if ver:
+                    warn += ' {}'.format(ver)
+                warn += '\n'
```

Comment:
You could do it like this:

```
version_restriction = ''
if opt == 'CONFIG_CC_STACKPROTECTOR_STRONG':
    version_restriction = ' (4.1.x and later versions only)'
elif opt == 'CONFIG_DEVPTS_MULTIPLE_INSTANCES':
    version_restriction = ' (4.8.x and earlier versions only)'
warn += '{}{}\n'.format(opt, version_restriction)
```

A little shorter, but I'm just being a PITA, I'm ok with how it is now.